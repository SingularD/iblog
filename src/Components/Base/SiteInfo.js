import React from 'react'
import PropTypes from 'prop-types'


class SiteInfo extends React.Component{
  constructor(props) {
    super(props)
  }
  render() {
    return (
      <div className="site-info-container col-12 row
      justify-content-center align-items-center
      border-top pt-4
      ">
        <p>© 2019 - 225 的小站 - 京 ICP 备2016100104号</p>
        <p>Powered by Bootstrap & React JS</p>
      </div>
    )
  }
}

export default SiteInfo
