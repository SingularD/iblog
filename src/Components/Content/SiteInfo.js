import React from 'react'

import { siteInfo } from "../../Data";

class SiteInfo extends React.Component{
  constructor(props) {
    super(props)
    this.state = {
      siteInfoHeader: siteInfo.siteInfoHeader,
      siteInfoFooter: siteInfo.siteInfoFooter
    }
  }
  render() {
    return (
      <div className="site-info-container col-12 row
      justify-content-center align-items-center
      border-top pt-4
      ">
        <p className="site-info-header">{this.state.siteInfoHeader}</p>
        <p className="site-info-footer">{this.state.siteInfoFooter}</p>
      </div>
    )
  }
}

export default SiteInfo
