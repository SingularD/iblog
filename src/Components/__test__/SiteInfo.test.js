import React from 'react'
import { shallow } from 'enzyme/build'
import SiteInfo from '../Base/SiteInfo'

const info = [
  '© 2019 - 225 的小站 - 京 ICP 备2016100104号',
  'Powered by Bootstrap & React JS'
]

describe('测试网站页面最底部的网站信息与备案信息', () => {
  it('检测信息是否渲染', () => {
    const wrapper = shallow( <SiteInfo /> )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('p').at(0).text()).toEqual(info[0]);
    expect(wrapper.find('p').at(1).text()).toEqual(info[1])
  });
})
