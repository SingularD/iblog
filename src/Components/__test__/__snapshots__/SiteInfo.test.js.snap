// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试网站页面最底部的网站信息与备案信息 检测信息是否渲染 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SiteInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="site-info-header"
        >
          © 2019 - 225 的小站 - 京 ICP 备2016100104号
        </p>,
        <p
          className="site-info-footer"
        >
          Powered by Bootstrap & React JS
        </p>,
      ],
      "className": "site-info-container col-12 row justify-content-center align-items-center border-top pt-4 ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© 2019 - 225 的小站 - 京 ICP 备2016100104号",
          "className": "site-info-header",
        },
        "ref": null,
        "rendered": "© 2019 - 225 的小站 - 京 ICP 备2016100104号",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Powered by Bootstrap & React JS",
          "className": "site-info-footer",
        },
        "ref": null,
        "rendered": "Powered by Bootstrap & React JS",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="site-info-header"
          >
            © 2019 - 225 的小站 - 京 ICP 备2016100104号
          </p>,
          <p
            className="site-info-footer"
          >
            Powered by Bootstrap & React JS
          </p>,
        ],
        "className": "site-info-container col-12 row justify-content-center align-items-center border-top pt-4 ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "© 2019 - 225 的小站 - 京 ICP 备2016100104号",
            "className": "site-info-header",
          },
          "ref": null,
          "rendered": "© 2019 - 225 的小站 - 京 ICP 备2016100104号",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Powered by Bootstrap & React JS",
            "className": "site-info-footer",
          },
          "ref": null,
          "rendered": "Powered by Bootstrap & React JS",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
